src/openloco/input/mouse_input.cpp

```diff
@@ -1426,8 +1426,23 @@ namespace openloco::input
                 break;
             }
             case ui::scrollview::scroll_part::hscrollbar_button_right:
-                call(0x4c89ae, regs);
+            {
+                ui::window* w = WindowManager::find(_pressedWindowType, _pressedWindowNumber);
+                if (w != nullptr)
+                {
+                    w->scroll_areas[scrollAreaIndex].flags |= scroll_flags::HSCROLLBAR_RIGHT_PRESSED;
+                    int16_t widget_width = w->widgets[_pressedWidgetIndex].right - w->widgets[_pressedWidgetIndex].left - 1;
+                    if ((w->scroll_areas[scrollAreaIndex].flags & scroll_flags::VSCROLLBAR_VISIBLE) != 0)
+                    {
+                        widget_width -= SCROLLBAR_WIDTH + 1;
+                    }
+                    int16_t widget_content_width = std::max(w->scroll_areas[scrollAreaIndex].h_right - widget_width, 0);
+                    w->scroll_areas[scrollAreaIndex].h_left = std::min((int16_t)(w->scroll_areas[scrollAreaIndex].h_left + 3), widget_content_width);
```

Comment:
prefer static casts over c style casts. I think you can also do `std::min<int16_t>(w->scroll_areas[scrollAreaIndex].h_left + 3, widget_content_width);`